### DISCUSSION TITLE
Use the instatiation of the console object as a one place flag for turning off all console.log()

### DISCUSSION THREAD

**Discussion by Dario-Mantegazza** [Textualize/rich#1047] [Ideas] [answered]
Hi, 
I'm using rich with the console.log to debug and follow the processes that are running in my scripts. 
What I would like to have is a parameter for the console Class that, if set to True, would turn off all the console.log calls placed in my code.
Like this:
```
from rich.console import Console

console = Console(log_off=True)

main():
  print("hi")
  console.log("hello")

Output:
hi
```
The idea is that sometimes I need those logs and some other times I want a cleaner output on my terminal. Currently for what I know about this library, my only option is to comment out every line where I have a console.log(). 
I hope I expressed my idea well enough.
Thanks for the awesome lib.
Cheers
Dario

**Comment by willmcgugan** [comment, marked as answer]
Hi Dario, 

I'd recommend you use the [logging](https://docs.python.org/3/library/logging.html) module disabling log output. You can use Rich [logging handler](https://rich.readthedocs.io/en/latest/logging.html) to keep the styled output.

If you really can't live without disabling `console.log`, you could do this:

```python
console.log = lambda *args, **kwargs: None
```

**Reply by Dario-Mantegazza** [reply]
Thanks
